### PR TITLE
Devspaces support

### PR DESCRIPTION
**Pull request by mangelajo**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
 default: build
 
 docs: sync
 	cd docs && uv run make html
 
 serve-docs: sync
-	cd docs && uv run make serve
-
-watch-docs:
-	sphinx-autobuild docs/source docs/build/html
```
Comment (mangelajo) [inline Makefile:14]: serve-docs was already doing the sphinx-autobuild

**Comment by NickCao** [comment]
Is it possible to use devspace in github codespace? So we don't have to maintain two sets of dev environments.

**Comment by mangelajo** [comment]
> Is it possible to use devspace in github codespace? So we don't have to maintain two sets of dev environments.

Not yet, I have hope that all this will converge over time.

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -37,3 +37,6 @@ RUN chown -R vscode $CARGO_HOME
 
 # Install required tools for development
 RUN apt-get update && apt-get install -y iperf3 libusb-dev
+
+# this is a workaround from uv to enable using vscode launch/python debug
+RUN echo "alias python='uv run python'" >> /home/user/.bashrc
```
Comment (NickCao) [inline .devcontainer/Dockerfile:42]: A better solution is to configure vscode with a custom interpreter path, as in https://github.com/jumpstarter-dev/jumpstarter/blob/70717aef5824047de67bfa1183bbcd747d178d42/.devcontainer/devcontainer.json#L41

File: .devfile.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+schemaVersion: 2.3.0
+metadata:
+  name: jumpstarter-1hpi
+  namespace: che-kube-admin-devspaces-jhpf17
```
Comment (NickCao) [inline .devfile.yaml:4]: Can we not hardcode these?

File: .devfile.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+projects:
+  - name: jumpstarter
+    git:
+      remotes:
+        origin: https://github.com/jumpstarter-dev/jumpstarter.git
+components:
+  - name: runtime
+    container:
+      image: quay.io/jumpstarter-dev/jumpstarter-devspace:latest
```
Comment (NickCao) [inline .devfile.yaml:13]: What would happen if the image is out of sync with the git remote (e.g. working on a different branch other than main)

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (mangelajo) [reply .devcontainer/Dockerfile:42]: Oh, ok, This is not necessary on this Dockerfile then, I will check if there is an equivalent for devfiles. Thanks

File: .devfile.yaml (hunk 3 above)
Comment (mangelajo) [reply .devfile.yaml:4]: We shouldn't, removing it, thanks.

File: .devfile.yaml (hunk 4 above)
Comment (NickCao) [reply .devfile.yaml:13]: What about using an image component: https://devfile.io/docs/2.3.0/adding-an-image-component

**Review by NickCao — APPROVED** [review]

**Pull request merged**